src/codecs/Base.js

```diff
@@ -0,0 +1,67 @@
+// A Dataframe property Codec is used to transform between property value encodings.
+// Three different encodings are handled:
+// * The `source` encoding correspond to the format in which
+//   the property values appear in the data sources (e.g. in MVT)
+// * The `internal` encoding is used in Dataframe properties
+//   apt for GPU consumption.
+// * The `external` encoding is the format in which feature
+//   properties are presented to the user
+//   (e.g. `property.eval(feature)` or `globalMin(property)`)
+// The Codec base class serves both as a base for derived codecs.
+// Codecs are associated to source properties.
+// There are two kind of codecs:
+// * Scalar (simple) encoder: source value is encoded as a single internal value
+// * Range encoder: a source value encodes as a pair [lo, hi] of values.
```

Comment:
It would be nice to have the jsdoc already prepared for this group of codec classes

Reply:
Of course! I've been really lazy about that 😞